File: hyunjae/프로그래머스/고득점키트/가장_큰_수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        // int 배열을 Integer 배열로 변환(sort이용을 위해)
        Integer[] numbersArr = Arrays.stream(numbers).boxed().toArray(Integer[]::new);
        
        Arrays.sort(numbersArr, new Comparator<Integer>() {
            //각각 합쳐서 비교해봐서 앞에두면 더 큰 것을 앞으로오게 정렬
            @Override
            public int compare(Integer o1, Integer o2) {
                String str1 = o1.toString();
                String str2 = o2.toString();
                
                String temp1 = str1 + str2;
                String temp2 = str2 + str1;
                return temp2.compareTo(temp1);
            }
        });
        
        StringBuilder result = new StringBuilder();
        for (int num : numbersArr) {
            result.append(num);
        }
        
        // 0으로만 구성된 경우 처리
        if (result.toString().startsWith("0")) {
            return "0";
        }
        
        return result.toString();
    }
}
    

File: sejeong/가장큰수.py
def solution(numbers):
    numbers = list(map(str,numbers)) # 리스트의 숫자를 문자열로 바꾸기
    # 문자열의 비교는 첫자리부터 큰순서대로 비교하므로 숫자를 여러번 반복했을때 큰숫자가 오도록 정렬하기
    numbers = sorted(numbers,key=lambda x:x*4,reverse=True)
    answer = ''
    for i in numbers:
        answer += i
    if answer[0] == '0':  # 만약 '000' 이런식으로 될때는 0만출력해야함
        return '0'
    return answer


numbers = [6, 10, 2]
print(solution(numbers))


